README.md

```diff
@@ -83,12 +83,150 @@ Design and implement the feature considering the relationship between the Produc
+    - -> stored inside Order
+- Example Response
+`[
+  {
+    "orderId": 456,
+    "orderDateTime": "2025-08-09T14:35:00Z",
+    "status": "PAID",
+    "items": [
+      { "productName": "T-Shirt", "option": "Large", "quantity": 2 }
+    ],
+    "checkoutSessionId": "pi_3OKjdf9sjlkd09",
+    "amount": 3999,
+    "currency": "USD",
+    "paymentMethod": "visa"
+  }
+]
+`
+## Step 2.3 - Deployment
+You must deploy your existing service and ensure it can interact with the client.
+### Features
+- [x] Write a deployment script to automate the deployment process. (`deploy.sh`)
+- [x] Handle security issues when interacting with the client API.
+  - [x] For example, resolve issues caused when the server and client have different Origin values.
+- [ ] HTTPS is optional
+- [x] Cors Configuration #applyPermitDefaultValues()
+  - [x] Allow all origins.
+  - [x] Allow “simple” methods GET, HEAD and POST.
+  - [x] Allow all headers.
+  - [x] Set max age to 1800 seconds (30 minutes).
+- [x] Test with MockMVC
+  - [x] Test client → server calls in local & production.
+  - [x] Confirm no CORS errors in browser console.
+
+## Action Plan for Step 2
+1. [x] Design Order domain & DTOs (Order, OrderItem, enums, payment metadata).
+2. [x] Add repositories (or entities).
+3. [x] Add Stripe config + SDK dependency. (Use test/sandbox key in application.properties, never commit it.) Stripe Docs
+4. [x] Implement StripeService to create/confirm PaymentIntents (with idempotency). Stripe Docs+1
+5. [x] Implement OrderService.placeOrder() orchestration: validate stock, create PENDING order, call Stripe, finalize (decrement stock + cart cleanup) on success. Use DB locking & transactions when changing stock.
+6. [x] Implement error mapping & friendly messages for Stripe decline codes (expired_card, insufficient_funds, incorrect_cvc, card_declined, etc.). Stripe Docs+1
+7. Add webhook endpoint (/webhooks/stripe) to process payment_intent.succeeded / fallback reconciliation (recommended). Stripe Docs
+8. [x] Add controller endpoints: POST /orders/place and GET /orders (with pagination).
+9. [x] Add tests: unit tests (mock Stripe), integration tests (Stripe test keys or WireMock), MockMvc tests for controllers.
+10. [x] Deploy: script (deploy.sh), run migrations, restart service, test client-server flows (CORS config).
+
+
 ## Considerations
 
 - [x] remove Boolean return type from all delete methods
 - [ ] change Double to BigDecimal inside the Entity (Product/price)
 - [x] decide on where and how to use Models
     - Entity == Model
-- [ ] dont use Cascade.All but Persist, Merge etc.
+- [x] don't use Cascade.All but Persist, Merge etc.
 - [ ] move product-option mapping to the constructor
-- [ ] effective test: Create InMemory fake repos -> service uses the fake repos
\ No newline at end of file
+- [ ] effective test: Create InMemory fake repos -> service uses the fake repos
+- [ ] add HTTPS
+- [ ] add webhook endpoint (/webhooks/stripe)
```

Comment:
I see there are still some [left over considerations](https://github.com/herotechcourse/spring-ecommerce-order/pull/14#discussion_r2257163060) and even new ones as well! 😮 

Since we're now past the time of this mission, I think it would be great to remove these considerations to make it not feel like it's not finished. You've done a great job with this project already 👍 

(I found another unchecked box under `Orders` and `Features`)